### ISSUE TITLE
.ask tests should be run by default

### ISSUE THREAD

**Issue by czerwinskilukasz1** [xFAANG/askql#166]
@km4 wrote on [Discord channel](https://discord.com/channels/688128418914500838/704117328735895662/721036181147615284):
> if create test and write "foobarbaz" in test in *.ask file the test passed :stuck_out_tongue:

The reason is that the .ask test is _parsed only_.  It will be run only if an additional *.test.result.ts file with an expected result value is provided.

I also found it confusing and vote for running the .ask programs every time, no matter whether *.test.result.ts exists or not.
The existence of *.test.result.ts would just result in comparing the result value.

**Comment by mhagmajer** [comment]
Duplicate of https://github.com/xFAANG/askql/issues/154, https://github.com/xFAANG/askql/pull/114

**Comment by czerwinskilukasz1** [comment]
It's not a duplicate.

**Comment by mhagmajer** [comment]
Why do you think this is not a duplicate @czerwinskilukasz1? What new problems does this issue cover? Our test suit very explicitly says whether or not ask files were run. I have explained that already in https://github.com/xFAANG/askql/pull/114#issuecomment-643696740

**Comment by czerwinskilukasz1** [comment]
@mhagmajer , #154 is about adding new resources `test` and `expect` so that we can define test files entirely in AskScript.  
As for your comment in #114, which you mentioned, it exactly describes the counter-intuitive behavior which I propose to change in the current ticket, which is:
If the .ask test file is defined, it should be executed.
#114 is about lessThan operator not about running .ask test files though, so I created a new ticket here.

**Comment by mhagmajer** [comment]
Ok, I see. How about we don't report  "starts" and "compiles" tests if they are succesful and only do the "computes" test or show it as a todo for the entire suite?

**Comment by czerwinskilukasz1** [comment]
I am not sure what "starts" means, but "compiles" is an important test - it says that the parser & transpiler successfully created AskCode from AskScript, right?
So I support keeping both "compiles" and "computes".

**Comment by mhagmajer** [comment]
If we keep both compiles and computes and only compiles passes and computes is todo then the related .ask test suite will be shown as PASS ✅  as currently. I haven't checked by maybe if the only test is todo, then the entire suite is also show as TODO which could help with the confusion you have raised here.

**Comment by czerwinskilukasz1** [comment]
@mhagmajer, I think we are diverging from the point I wanted to make in this issue.
I am saying that if a developer provides an .ask test file, it might be completely incorrect and it will still pass if only it parses correctly from AskScript to AskCode, e.g. use undeclared identifiers, undeclared functions etc.
My proposal is to *execute* every ask test file and make sure it succeeds even if we don't know the expected output.

**Comment by mhagmajer** [comment]
How do you propose to execute AskScripts that expect arguments? We are also planning to be gradually migrating from TypeScript to AskScript in this repository. How do you envision this working then?

**Comment by czerwinskilukasz1** [comment]
> We are also planning to be gradually migrating from TypeScript to AskScript in this repository.

Who is "we" in "We are also planning" if we two haven't talked about it neither did we discussed any Github issues related to this topic?

**Comment by mhagmajer** [comment]
> Who is "we" in "We are also planning" if we two haven't talked about it nor did we discussed any Github issues related to this topic, nor have it seen in milestones?

If AskScript is better than TypeScript+GraphQL than it would be beneficial for us to migrate. I remember bringing this up already.

What is actionable here except https://github.com/xFAANG/askql/issues/154?

**Comment by mhagmajer** [comment]
I have that checked that Jest shows test suite as PASS even if its only test is "todo"

**Comment by czerwinskilukasz1** [comment]
> > Who is "we" in "We are also planning" if we two haven't talked about it nor did we discussed any Github issues related to this topic, nor have it seen in milestones?
> 
> If AskScript is better than TypeScript+GraphQL than it would be beneficial for us to migrate. I remember bringing this up already.
> 
> What is actionable here other than #154?

@mhagmajer , please read my previous messages from this ticket.

**Comment by mhagmajer** [comment]
@czerwinskilukasz1 yes, I have red it and I still don't understand why you think that the test suite should run all .ask files in this repository or how would it do it - some programs require arguments. Very soon part of our source code (including tests) will be in AskScript itself.

**Comment by czerwinskilukasz1** [comment]
> @czerwinskilukasz1 yes, I have red it and I still don't understand why you think that the test suite should run all .ask files in this repository or how would it do it - some programs require arguments. 

I am not sure what you mean as we already have a mechanism for providing arguments to .ask programs: `.test.args.ts` files.
And I doubt people would add new tests that require arguments without providing them.

> Very soon part of our source code (including tests) will be in AskScript itself.

When were you planning to do the conversion?
